File: 5. Doesn't He Have Intern-Elves For This/2.Solution.py
ans = 0

for i in range(1000):
    s = input()
    last1 = ''
    last2 = ''
    pair = False
    reap = False
    for k in range(len(s)):
        j = s[k]
        if j == last2:
            reap = True
        if last1 and last1+j in s[k+1:]:
            pair = True
        if reap and pair:
            ans += 1
            break
        last2 = last1
        last1 = j

print(ans)

File: 2016/9. Explosives in Cyberspace/2.Solution.cpp
#include <bits/stdc++.h>

using namespace std;

int main() {
    ifstream input("input.txt");
    char ch;
    long long ans = 0;
    stack<pair<int, long long>> rewards;
    int where = 0;
    while (input.get(ch)) {
        if (ch == '(') {
            where++;
            string num1;
            string num2;
            while (where++ && input.get(ch) && ch != 'x') {
                num1.push_back(ch);
            }
            while (where++ && input.get(ch) && ch != ')') {
                num2.push_back(ch);
            }
            // store how much character does it count until "where"
            // for example (3x2) means 3 characters from now (3 + where) will have 2 duplicates each
            long long rew = stoi(num2);
            int togo = stoi(num1) + where;
            if (!rewards.empty() && rewards.top().first >= togo) {
                rewards.push({togo, rew * rewards.top().second});
            } else {
                rewards.push({togo, rew});
            }
            
        } else {
            while (!rewards.empty() && rewards.top().first < where) rewards.pop();
            if (rewards.empty()) ans++;
            else {
                ans += rewards.top().second;
            }
            where++;
            while (!rewards.empty() && rewards.top().first <= where) rewards.pop();
        }
    }
    cout << ans;
}
